src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -310,4 +383,28 @@ private function supplyTranslatedContentName(Node $node, array $contentById): vo
             $this->supplyTranslatedContentName($child, $contentById);
         }
     }
+
+    /**
+     * @param array|null $aggregationResult
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function supplyChildrenCount(Node $node, array $aggregationResult = null): void
```

Comment:
It would be cleaner if you used empty array instead of `null` value.

Reply:
Passing `null` here indicates that Aggregation results are not available (for example: using a legacy search engine) and the "legacy" way of counting children should be used inside the `supplyChildrenCount` method. An empty array is a totally valid result of aggregation. I can change it to pass to an empty array, but we will have to introduce something like `$isAggregationSupported` and pass it alongside to `supplyChildrenCount`

Reply:
>> It would be cleaner if you used empty array instead of `null` value.

> Passing `null` here indicates that Aggregation results are not available (for example: using a legacy search engine) and the "legacy" way of counting children should be used inside the `supplyChildrenCount` method. An empty array is a totally valid result of aggregation. I can change it to pass to an empty array, but we will have to introduce something like `$isAggregationSupported` and pass it alongside to `supplyChildrenCount`

@mateuszbieniek I don't see how distinguishing between `[]` and `null` would change the behavior of this method in its current state. Could you show a snippet with `[]` instead of `null`?

Reply:
There are two possibilities:
1. Aggregations are not supported (`$aggregationResult` is `null`). Every single container node will have `$this->countSubitems($node->locationId);` called

2. Aggregations are supported and `$aggregationResult` is an array.
```
$totalCount = isset($aggregationResult[$node->locationId]) ?
                    $aggregationResult[$node->locationId] :
                    0;
```
If `$aggregationResult` has key `$node->locationId` then node has children, if a key does not exist, this means that there children count is `0`;

So, if I pass it as empty array, when **aggregations are not supported** current state of method will result on 0 children count for all nodes.

Reply:
I can change it to an empty array but will have to add a check for aggregation support inside `supplyChildrenCount` method.

Reply:
I've changed the check inside from `supplyChildrenCount`:
```
if ($aggregationResult) {
```
to
```
if ($aggregationResult !== null) {
```
This should reduce the confusion.